src/main/java/core/basesyntax/impl/StorageImpl.java

```
@@ -3,17 +3,53 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_ITEMS_NUMBER];
+        values = (V[]) new Object[MAX_ITEMS_NUMBER];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size < MAX_ITEMS_NUMBER) {
+            int index = findKeyIndex(key);
+            if (index != -1) {
+                values[index] = value;
+            } else {
+                keys[size] = key;
+                values[size] = value;
+                size++;
+            }
+        } else {
+            System.out.println("The storage is full, it is impossible to make a new entry");
```

For the future, it is not good practice to log data using such a way,
It is better to throw an Exception or use the logger
```suggestion
```